examples/2_Intermediate/stage_two_optimization_stochastic.py

```diff
@@ -0,0 +1,191 @@
+
+The coil perturbations for each coil are the sum of a 'systematic error' and a
+'statistical error'.  The former satisfies rotational and stellarator symmetry,
+the latter is independent for each coil.
+"""
+
+from simsopt.geo.surfacerzfourier import SurfaceRZFourier
+from simsopt.objectives.fluxobjective import SquaredFlux, CoilOptObjective
+from simsopt.geo.curve import RotatedCurve, curves_to_vtk, create_equally_spaced_curves
+from simsopt.field.biotsavart import BiotSavart
+from simsopt.field.coil import Current, Coil, coils_via_symmetries
+from simsopt.geo.curveobjectives import CurveLength, MinimumDistance
+from simsopt.geo.curveperturbed import GaussianSampler, CurvePerturbed, PerturbationSample
+import numpy as np
+import os
+from pathlib import Path
+
+# Number of unique coil shapes, i.e. the number of coils per half field period:
+# (Since the configuration has nfp = 2, multiply by 4 to get the total number of coils.)
+ncoils = 4
+
+# Major radius for the initial circular coils:
+R0 = 1.0
+
+# Minor radius for the initial circular coils:
+R1 = 0.5
+
+# Number of Fourier modes describing each Cartesian component of each coil:
+order = 5
+
+# Weight on the curve lengths in the objective function:
+ALPHA = 1e-6
+
+# Threshhold for the coil-to-coil distance penalty in the objective function:
+MIN_DIST = 0.1
+
+# Weight on the coil-to-coil distance penalty term in the objective function:
+BETA = 10
+
+# Standard deviation for the coil errors
+SIGMA = 1e-3
+
+# Length scale for the coil errors
+L = 0.5
+
+# Number of samples to approximate the mean
+N_SAMPLES = 16
+
+# Number of samples for out-of-sample evaluation
+N_OOS = 256
+
+# Number of iterations to perform:
+ci = "CI" in os.environ and os.environ['CI'].lower() in ['1', 'true']
+MAXITER = 50 if ci else 400
+
+# File for the desired boundary magnetic surface:
+TEST_DIR = (Path(__file__).parent / ".." / ".." / "tests" / "test_files").resolve()
+filename = TEST_DIR / 'input.LandremanPaul2021_QA'
+
+#######################################################
+# End of input parameters.
+#######################################################
+
+# Initialize the boundary magnetic surface; errors break symmetries, so consider the full torus
+nphi = 64
+ntheta = 16
+s = SurfaceRZFourier.from_vmec_input(filename, range="full torus", nphi=nphi, ntheta=ntheta)
+
+# Create the initial coils:
+base_curves = create_equally_spaced_curves(ncoils, s.nfp, stellsym=True, R0=R0, R1=R1, order=order)
+base_currents = [Current(1e5) for i in range(ncoils)]
+# Since the target field is zero, one possible solution is just to set all
+# currents to 0. To avoid the minimizer finding that solution, we fix one
+# of the currents:
+base_currents[0].fix_all()
+
+coils = coils_via_symmetries(base_curves, base_currents, s.nfp, True)
+bs = BiotSavart(coils)
+bs.set_points(s.gamma().reshape((-1, 3)))
+
+curves = [c.curve for c in coils]
+curves_to_vtk(curves, "/tmp/curves_init")
```

Comment:
On clusters, I noticed this line gives an error if users don't have permission to write to `/tmp`. This might be confusing or irritating to a new user. Any objection if `/tmp/` is removed throughout this script and `stage_two_optimization.py` so the files are just written to the working directory? Or to a subdirectory of `examples/2_Intermediate`?